src/lavinmq/auth/authenticators/basic.cr

```diff
@@ -0,0 +1,21 @@
+require "../authenticator"
+require "../../server"
+
+module LavinMQ
+  module Auth
+    class BasicAuthenticator < Authenticator
+      def initialize(@users : UserStore)
+      end
+
+      def authenticate(username : String, password : String)
+        begin
+          user = @users[username]
+          return user if user && user.password && user.password.not_nil!.verify(password)
+          Log.info { "Basic authentication failed" }
```

Comment:
I think you can remove this. The `ConnectionFactory` will log failed authentications with more details.